index.js

```diff
@@ -12,6 +12,20 @@ var DEFAULT_PORTS = {
 	'ftp:': 21
 };
 
+// protocols that always contain a // bit.
+var slashedProtocol = {
+	'http': true,
+	'https': true,
+	'ftp': true,
+	'gopher': true,
+	'file': true,
+	'http:': true,
+	'https:': true,
+	'ftp:': true,
+	'gopher:': true,
+	'file:': true
+};
```

Comment:
Any around having to hardcode this? Maybe we could detect slashed protocol by putting it through `url.parse` and checking the result?